lib/active_record/connection_adapters/spanner/database_statements.rb

```diff
@@ -220,20 +227,33 @@ def create_delete_all_mutation arel
           unless arel.ast.wheres.empty?
             raise "A delete mutation can only be created without a WHERE clause"
           end
+          table_name = arel.ast.relation.name if arel.ast.relation.is_a? Arel::Table
+          table_name = arel.ast.relation.left.name if arel.ast.relation.is_a? Arel::Nodes::JoinSource
+          unless table_name
+            raise "Could not find table for delete mutation"
+          end
 
           Google::Cloud::Spanner::V1::Mutation.new(
             delete: Google::Cloud::Spanner::V1::Mutation::Delete.new(
-              table: arel.ast.relation.name,
+              table: table_name,
               key_set: { all: true }
             )
           )
         end
 
-        DDL_REGX = ActiveRecord::ConnectionAdapters::AbstractAdapter
-                   .build_read_query_regexp(:create, :alter, :drop).freeze
+        COMMENT_REGEX = %r{(?:--.*\n)*|/\*(?:[^*]|\*[^/])*\*/}m.freeze \
+            unless defined? ActiveRecord::ConnectionAdapters::AbstractAdapter::COMMENT_REGEX
+        COMMENT_REGEX = ActiveRecord::ConnectionAdapters::AbstractAdapter::COMMENT_REGEX \
+            if defined? ActiveRecord::ConnectionAdapters::AbstractAdapter::COMMENT_REGEX
```

Comment:
The constant `ActiveRecord::ConnectionAdapters::AbstractAdapter::COMMENT_REGEX` is not defined in all 6.x versions, so we need to define it in the Spanner adapter when it is not defined in ActiveRecord.